### PR TITLE
Fix AI responses visually appending to previous response

### PR DESCRIPTION
**Pull request by adeeshaek**
## Summary
- Harden `shouldSuppressDuplicateResultMessage` to treat prior `result` messages as turn boundaries, preventing the backward scan from crossing into previous turns during reconnect/replay edge cases
- Add `filterDuplicateResultMessages` as a rendering-level safety net that strips duplicate result text before message grouping, so even if store-level dedup misses an edge case the duplicate won't render
- Wire the filter into the chat content `useMemo` pipeline before `groupAdjacentToolCalls`

## Test plan
- [x] New test: result-as-turn-boundary in `protocol.test.ts` (verifies current-turn dedup still works + cross-turn dedup is blocked)
- [x] New tests: 5 cases for `filterDuplicateResultMessages` in `chat-protocol.test.ts` (duplicate filtered, different text kept, empty text kept, no preceding assistant kept, user boundary respected)
- [x] All 2787 existing tests pass
- [x] `pnpm typecheck` clean
- [x] `pnpm check:fix` clean
- [x] Pre-commit hooks pass (lint, typecheck, dep-cruiser, knip)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/chat-protocol.ts [hunk 1]
```diff
@@ -543,6 +543,48 @@ function extractPairedToolCalls(toolMessages: ChatMessage[]): PairedToolCall[] {
...
+  for (let i = startIndex - 1; i >= 0; i -= 1) {
+    const prev = messages[i];
+    if (!prev) {
+      continue;
+    }
+    if (prev.source === 'user') {
+      return null;
+    }
+    if (prev.source !== 'agent' || !prev.message || prev.message.type === 'result') {
+      continue;
```
[the remark this answers is not in the document]
Comment (adeeshaek) [reply src/lib/chat-protocol.ts]: @greptile-apps - Fixed as suggested. `findPrecedingAgentText` now returns `null` when it encounters a `result` message, treating it as a turn boundary consistent with `shouldSuppressDuplicateResultMessage`. Added a test case verifying the backward scan doesn't cross result boundaries.
